File: app/src/main/java/com/pgssoft/mvvm/services/interfaces/Repository.java
package com.pgssoft.mvvm.services.interfaces;


import com.pgssoft.mvvm.model.api.ApiTable;
import com.pgssoft.mvvm.model.database.Rate;
import com.pgssoft.mvvm.model.database.Table;

import java.util.List;

/**
 * Created by bstokrocki on 30.01.2017.
 */
public interface Repository {
    void saveTables(ApiTable[] apiTables);

    List<Rate> getAllRates();

    Rate findRateById(String rateId);
}

File: kotlin/app/src/main/java/com/pgssoft/kotlinplayground/service/iface/Repository.kt
package com.pgssoft.kotlinplayground.service.iface


import com.pgssoft.kotlinplayground.model.RateDto
import com.pgssoft.kotlinplayground.model.api.ApiTable
import com.pgssoft.kotlinplayground.model.db.Rate

/**
 * Created by bstokrocki on 30.01.2017.
 */
interface Repository {
    fun saveTables(apiTables: Array<ApiTable>)

    fun getAllRates(): List<Rate>

    fun findRateById(rateId: String): RateDto?
}
